#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, bool_t, double_t, str_t

gen = ParameterGenerator()
gen.add("enabled", bool_t, 0, "Whether to apply this plugin or not", True)
gen.add("update_zone_enlargment", double_t, 0, "size in meter the update area is enlarged ro ensure old moving objects are completely removed", 0.5, 0.0, 3.0)
gen.add("own_unique_name", str_t, 0, "own unique name in moving_object_msg, will than be ignored as obstacle")
gen.add("obstacle_life_time", double_t, 0, "delete objects from obstacle store if last message is older than x seconds", 3.0, 0.0, 30.0)
exit(gen.generate("virtual_obstacles", "virtual_obstacles", "GenericPlugin"))
